MapleServer2/Servers/Game/FieldManager.cs

```diff
@@ -396,5 +386,39 @@ private Task StartHealingSpot(GameSession session, IFieldObject<Player> player)
                 }
             });
         }
+
+        private void SpawnMobs(IFieldObject<MobSpawn> mobSpawn)
+        {
+            List<CoordF> spawnPoints = MobSpawn.SelectPoints(mobSpawn.Value.SpawnRadius);
+
+            foreach (NpcMetadata mob in mobSpawn.Value.SpawnMobs)
+            {
+                int spawnCount = mob.NpcMetadataBasic.GroupSpawnCount;  // Spawn count changes due to field effect (?)
+                if (spawnCount > mobSpawn.Value.MaxPopulation)
+                {
+                    break;
+                }
+
+                for (int i = 0; i < spawnCount && i < mobSpawn.Value.MaxPopulation; i++)
+                {
+                    IFieldObject<Mob> fieldMob = RequestFieldObject(new Mob(mob.Id, mobSpawn));
+                    fieldMob.Coord = mobSpawn.Coord + spawnPoints[mobSpawn.Value.Mobs.Count % spawnPoints.Count];
+                    AddMob(fieldMob);
+                }
+            }
+        }
+
+        private Task StartSpawnTimer(IFieldObject<MobSpawn> mobSpawn)
+        {
+            return Task.Run(async () =>
+            {
+                await Task.Delay(mobSpawn.Value.SpawnData.SpawnTime * 1000);
```

Comment:
Might be wrong but if you move this to before the return line, it only needs to do this multiplication one time

Reply:
I think it still does the multiplication one time since the function only runs once.

If you mean if the compiler would make it a constant, I think it would still recognize it as a constant in this case too, but I'm not sure as well. I can change it.